### PR TITLE
error handling v1

### PR DESCRIPTION
**Pull request by mini-wizbit**
version 1 of error handling

### CODE REVIEW

File: nc-games/src/components/AllReviews/AllReviews.jsx [hunk 1]
```diff
@@ -10,17 +10,25 @@ export const AllReviews = () => {
...
 
   useEffect(() => {
     setLoading(true);
-    getReviews(theCategory).then((ApiReviews) => {
-      setLoading(false);
-      setReviews(ApiReviews.reviews);
-    });
+    getReviews(theCategory)
+      .then((ApiReviews) => {
+        setIsError(true);
```
Comment (pkaodev) [inline nc-games/src/components/AllReviews/AllReviews.jsx]: 19 - why are we setting error to be true for a successful api request?

File: nc-games/src/components/AllReviews/AllReviews.jsx [hunk 2]
```diff
@@ -10,17 +10,25 @@ export const AllReviews = () => {
...
+        setLoading(false);
```
Comment (pkaodev) [inline nc-games/src/components/AllReviews/AllReviews.jsx:20]: probably best to setLoading after setReviews, but it wouldn't make much of a difference really

File: nc-games/src/components/AllReviews/AllReviews.jsx [hunk 3]
```diff
@@ -10,17 +10,25 @@ export const AllReviews = () => {
...
+      })
+      .catch((error) => {
+        setIsError(true);
+      });
   }, [theCategory]);
 
   return loading ? (
     <p>...Loading please wait</p>
+  ) : isError ? (
+    <p>Oops... something went wrong.</p>
```
Comment (pkaodev) [inline nc-games/src/components/AllReviews/AllReviews.jsx]: for now this is good enough (handling all errors as the same thing).  later on if you want to add more, you could handle different errors differently (depending on what the status code/msg is on the error object).

**Comment by pkaodev** [comment]
I am getting an error when I try to press Game Reviews (sent you an image of this).

Also if I go to a non-existent review (/reviews/11111) I get stuck on a loading message instead of seeing an error message.

**Comment by pkaodev** [comment]
Submit another PR when all of this has been fixed

File: nc-games/src/components/AllReviews/AllReviews.jsx (hunk 3 above)
Comment (mini-wizbit) [reply nc-games/src/components/AllReviews/AllReviews.jsx]: DONE

File: nc-games/src/components/AllReviews/AllReviews.jsx (hunk 1 above)
Comment (mini-wizbit) [reply nc-games/src/components/AllReviews/AllReviews.jsx]: Fixed that

**Comment by pkaodev** [comment]
Getting error messages instead of comments, and when I add my own comment.  I have sent you screenshots of these.

I *think* it might have to do with where you are using setError(true)

**Review by hevdevs — APPROVED** [review]
Looks good to merge based on previous suggestions!

**Pull request merged**